reporter.go

```diff
@@ -38,3 +39,28 @@ func NewRequireReporter(t require.TestingT) *RequireReporter {
 func (r *RequireReporter) Errorf(message string, args ...interface{}) {
 	r.backend.FailNow(fmt.Sprintf(message, args...))
 }
+
+// FatalReporter is a struct that implements the testing.Reporter interface
+// and calls t.Fatalf() when a test fails.
+type FatalReporter struct {
+	backend testing.TB
+}
+
+// Errorf implements Reporter.Errorf.
+func (r *FatalReporter) Errorf(message string, args ...interface{}) {
+	r.backend.Fatalf(message, args...)
+}
+
+// Report is a method that takes a testing.TB interface and checks if the test
+// has failed by calling the Failed() method. If the test has failed, it calls
+// the Errorf method with the provided testing.TB interface and an error message.
+func (r *FatalReporter) Report(result testing.TB) {
+	if result.Failed() {
```

Comment:
Why do we need this method? When is it used?

Reply:
it was about this comment in the issue "<If you don't use testify, and want fatal assertions, you have to write your own reporter that invokes t.Fatalf. It would be nice to fill this gap and provide FatalReporter that does that for you - provides Errorf method that just invokes t.Fatalf. >"

The Report method in the FatalReporter struct is used to check the status of a test and take appropriate action based on the result. It takes a testing.TB interface, which represents a test or benchmark, as its argument. It then checks if the test has failed by calling the Failed() method of the testing.TB interface. If the test has failed, it calls the Errorf method with the provided testing.TB interface and an error message, which will stop the execution of the test suite immediately, preventing any further tests from running.

The Report method is typically called by the Go testing framework at the end of a test or benchmark, after the test function has completed execution. It is used to report the final result of the test or benchmark and take any necessary action based on whether the test passed or failed but also  the Report method is only called when a test or benchmark has completed execution, so if a test fails due to a call to t.Fatalf() or t.FailNow(), the Report method will not be called.

Reply:
Sorry, I don't get it.

> The Report method is typically called by the Go testing framework at the end of a test or benchmark, after the test function has completed execution.

Are you saying that `testing` package from stdlib calls Report method? How is it supposed to know about our reporter object? I don't see any mention of it in docs.

Also I'm completely missing motivation for this method.

If you do this:

```
r := NewFatalReporter(t)
r.Report(t)
```

it will be equivalent to:

```
if t.Failed() {
    t.Fatalf("Test failed")
}
```

Why do we ever need this? Maybe I'm missing something?

Reply:
Ah ok I see,  so I can remove I thought it will be the way to implement our own reporter that invoke Fatalf as said in the issue comment. I will remove it